constrained_by_test.go

```diff
@@ -0,0 +1,40 @@
+package worksheets
+
+import (
+	"strings"
+
+	"github.com/stretchr/testify/require"
+)
+
+func (s *Zuite) TestWorksheet_constrainedBy() {
+	defs, err := NewDefinitions(strings.NewReader(`worksheet simple {
+		1:name text constrained_by { return name == "Alex" || name == "Wilson" }
+	}`))
+	require.NoError(s.T(), err)
+
+	ws := defs.MustNewWorksheet("simple")
+
+	require.False(s.T(), ws.MustIsSet("name"))
+	err = ws.Set("name", NewText("Alice"))
+	require.Equal(s.T(), "\"Alice\" not a valid value for constrained field name", err.Error())
+	require.False(s.T(), ws.MustIsSet("name"))
+
+	err = ws.Set("name", NewText("Alex"))
+	require.NoError(s.T(), err)
+	require.True(s.T(), ws.MustIsSet("name"))
+	require.Equal(s.T(), `"Alex"`, ws.MustGet("name").String())
+}
+
+func (s *Zuite) TestWorksheet_constrainedByNonsensicalExpression() {
```

Comment:
How is `some_field + 2` non-sensical? Do you mean it's not a bool? Or do you mean it's a weird cyclic "constraint"?

Reply:
not a bool